user-service/index.js

```diff
@@ -1,28 +1,42 @@
 import express from 'express';
 import cors from 'cors';
 import cookieParser from 'cookie-parser';
-import { createUser, deleteAccount, logout, signin } from './controller/user-controller.js';
+import {
+  createUser,
+  updateAccount,
+  deleteAccount,
+  logout,
+  signin,
+  acknowledgeJWTValidity
+} from './controller/user-controller.js';
 import { authenticateJwt } from './model/jwt.js';
 
 const app = express();
-app.use(express.urlencoded({ extended: true }))
-app.use(express.json())
-app.use(cors()) // config cors so that front-end can use
-app.options('*', cors())
-app.use(cookieParser())
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+app.use(
+  cors({
+    origin: ['http://localhost:3000'],
+    credentials: true,
+  })
+); // config cors so that front-end can use
+app.options('*', cors());
+app.use(cookieParser());
 
-const router = express.Router()
+const router = express.Router();
 
 // Controller will contain all the User-defined Routes
-router.get('/', (_, res) => res.send('Hello World from user-service'))
-router.post('/', createUser)
-router.post('/signin', signin)
-router.post('/logout', authenticateJwt, logout)
-router.post('/delete', authenticateJwt, deleteAccount)
+router.get('/', (_, res) => res.send('Hello World from user-service'));
+router.post('/', createUser);
+router.post('/signin', signin);
+router.post('/logout', authenticateJwt, logout);
+router.post('/update', authenticateJwt, updateAccount);
+router.post('/delete', authenticateJwt, deleteAccount);
+router.get('/verify-jwt', authenticateJwt, acknowledgeJWTValidity);

```

Comment:
May I ask, so when you access the PrivateRoute component, it will run axios get on this backend URL to perform the check if a user is authenticated?

Reply:
yes, that's correct